internal/blockchain/ethereum/ethereum.go

```diff
@@ -726,6 +756,17 @@ func (e *Ethereum) InvokeContract(ctx context.Context, nsOpID string, signingKey
 	if err != nil {
 		return err
 	}
+	if batch != nil {
+		if err := e.checkDataSupport(ctx, abi); err != nil {
+			return err
+		}
+		method, batchPin := e.buildBatchPinInput(ctx, 2, "", batch)
```

Comment:
Note that we don't know the shape of the batch pin contract the user is ultimately going to call from their contract on-chain. We assume "version 2" of the ABI arguments (which has 4 args, where "version 1" expected 5 args) - but if we ever add a different version of this pinBatch method, there may need to be a way to choose upfront how to have the args encoded.